ghcide/session-loader/Development/IDE/Session.hs

```diff
@@ -479,12 +479,12 @@ loadSessionWithOptions SessionLoadingOptions{..} dir = do
 -- This then builds dependencies or whatever based on the cradle, gets the
 -- GHC options/dynflags needed for the session and the GHC library directory
 
-cradleToOptsAndLibDir :: Show a => Cradle a -> FilePath
+cradleToOptsAndLibDir :: Show a => Logger -> Cradle a -> FilePath
                       -> IO (Either [CradleError] (ComponentOptions, FilePath))
-cradleToOptsAndLibDir cradle file = do
+cradleToOptsAndLibDir logger cradle file = do
     -- Start off by getting the session options
     let showLine s = hPutStrLn stderr ("> " ++ s)
```

Comment:
The output of running cradle shouldn't be dumped to stderr as well -- test case `pluginSimpleTests` now reports:
```
simple plugin:
> Resolving dependencies...
> Build profile: -w ghc-8.10.4 -O1
> In order, the following will be built (use -v for more details):
>  - plugin-1.0.0 (lib) (first run)
> Configuring library for plugin-1.0.0..
> Warning: Packages using 'cabal-version: >= 1.10' and before 'cabal-version:
> 3.4' must specify the 'default-language' field for each component (e.g.
> Haskell98 or Haskell2010). If a component uses different languages in
> different modules then list the other ones in the 'other-languages' field.
> Preprocessing library for plugin-1.0.0..

OK (3.83s)
```
But considering that `showLine` is emitted per line: https://github.com/mpickering/hie-bios/blob/2969b2d26c325888c9fd51fea95478c70f657007/src/HIE/Bios/Cradle.hs#L843, I don't think send each line over lsp logger is appropriate.

Reply:
> It seems that `ghcide-tests: <file descriptor: 14>: hPutBuf: resource vanished (Broken pipe)` and `lsp:Got EOF, exiting 1 ...` are messing up the test report:

I think those lines denote tests that failed because of the `SQLError` and the retry is kicking in. That is good. 

Even if the test report is being a little dirty, I prefer not to silence the output since that will make test debugging much harder. Imho the silencing should be done by the test framework, and then the full output of test failures included at the end of the test report. I'm not a tasty expert so I don't know whether it supports this feature, either directly or via some plugin.